### PR TITLE
Remap delete action to 'D' key

### PR DESCRIPTION
**Pull request by vahron**
Hey team, I'm using Excalidraw extensively in the last couple of months and it's great! I'm amazed by performance boost it provides for diagram manipulations.

I noticed that some inconvenience when using editor with touchpad or mouse. Hotkey layout for some actions are out of reach for left hand. For example "Delete - backspace", "Send Front - cmd+]" or "Curved Line - L". 

Maybe it would be more convenient if all actions are actually located in reach of left hand (see picture). For example, map deletion on "D" and remap diamond shape to "G".

https://ibb.co/tHmXy4n
https://ibb.co/3RCkvDc
https://ibb.co/WHFZvhP

### CODE REVIEW

File: src/shapes.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ export const SHAPES = [
       </svg>
     ),
     value: "diamond",
-    key: "d",
+    key: "g",
```
Comment (vjeux) [inline src/shapes.tsx:34]: Is there a reason why "g", it doesn't anywhere in "diamond"

Comment (vahron) [reply src/shapes.tsx:34]: no particular reasons, just picked free letter in the left part of keyboard

**Comment by vahron** [comment]
fixed tests

**Comment by vahron** [comment]
fix tests 2

File: src/shapes.tsx (hunk 1 above)
Comment (vahron) [reply src/shapes.tsx:34]: I see other options are N and M possibly. They're both on the right side of keyboard , but relatively close to left still.

**Comment by vjeux** [comment]
cc @dwelle, @lipis for thoughts

**Comment by dwelle** [comment]
Personally, I'm not too keen on this change:

- We use letter-only keys almost exclusively for tools, and now we'd use `D` for a destructive action. Moreover, the `D` is surrounded by tool keys all around, so when people switch between tools, they'll inadvertently hit delete:

    image
- `D` for delete doesn't have precedent in any app I know of. If anything, `D` is used by Photoshop to switch to default color scheme (B/W).

As for the `L`, that has precedent at least in Figma. And the `+]` shortcuts have precedent in most vector apps.

**Comment by vahron** [comment]
Does Excalidraw have statistics on action triggers? For my personal experience, deletion action happens way more often compared to Diamond tool usage. I understand we want to keep scheme consistent with other products, but other products are meant to be optimized for user actions. 

I can suggest alternative "opt+D" for deletion action. I see "cmd+D" does duplicate, so does not break "tool-action" border. However, some editors actually have deletion as a tool.

**Comment by dwelle** [comment]
> Does Excalidraw have statistics on action triggers? For my personal experience, deletion action happens way more often compared to Diamond tool usage. I understand we want to keep scheme consistent with other products, but other products are meant to be optimized for user actions.

In this case it's not about usage. Even if we removed the diamond tool, we wouldn't want to map Delete to `D` for the aforementioned reasons.

Anyway, I don't see much benefit in mapping well-known Delete/Backspace shortcut to keys like `Option + D` (this combination might actually not work on mac keyboards as it may be mapped to some symbol --- dunno, don't have access to Mac), because it's not intuitive. For people to discover that, they'd need to search for it in the shortcut dialog --- as opposed to Delete which is intuitive and well established.

I suggest you use either:

- Use `Cmd + X` which *cuts* the element, but if you don't care about your clipboard, it should work the same a delete.
- Remap Delete key in your OS to something else, such as Caps Lock.

**Comment by vahron** [comment]
> Even if we removed the diamond tool, we wouldn't want to map Delete to D for the aforementioned reasons.

I see, is hotkey layout modelled after Figma? I agree with the argument on intuitive layout, but some actions can't rely purely on intuition. Hotkeys are used to speedup things for amateur users and their layout might be affected by usage frequence, even keyboard layout is build with this principle.

For example, there're three ways of element duplication: `opt+drag`, `ctrl+c, ctrl+v` and `cmd+d`. In the same time element deletion has only one hotkey. I can see `opt+d` can be used to balance out this hotkey map.

Anyway, I don't have usage data to support this point of view. I'm fine using `cmd+x` as deletion.

**Comment by vjeux** [comment]
Looks like ctrl+D on Keynote does deletion. That's probably a good addition.

**Comment by dwelle** [comment]
> Looks like ctrl+D on Keynote does deletion. That's probably a good addition.

Ok, we can do that. But it will be Mac only, for obvious reasons.

**Comment by dwelle** [comment]
@vahron you can rewrite this PR to use the `Ctrl+X`. I'm on mac, so can't test. Also, make sure the keybinding hint is listed only for Darwin platform.

**Comment by lipis** [comment]
Just a friendly ping @vahron :)

**Comment by dwelle** [comment]
Looks like there's not much interest in doing this. Anyone with a Mac, feel free to implement in new PR (or reopen this one) if you want so.

**Pull request closed without merging**